From 8dd5577174fe23ace5ecba7a81d2516315a0b301 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 30 Oct 2024 10:16:35 +0000
Subject: [PATCH] lib: OVF: Add preliminary support for Windows Server 2025

Unfortunately support has not yet been added to oVirt.  However I have
inferred the correct ostype value based on existing entries.

Also pull in the following commit from the common submodule:

  Richard W.M. Jones (1):
      mlcustomize: Add heuristic support for Windows Server 2025

Related: https://issues.redhat.com/browse/RHEL-65009
Related: https://issues.redhat.com/browse/RHEL-65010
Reported-by: Ming Xie
---
 common            | 2 +-
 lib/create_ovf.ml | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

Submodule common e9eea65a..2d42128e:
diff --git a/common/mlcustomize/inject_virtio_win.ml b/common/mlcustomize/inject_virtio_win.ml
index 88c1c8cf..7e46f318 100644
--- a/common/mlcustomize/inject_virtio_win.ml
+++ b/common/mlcustomize/inject_virtio_win.ml
@@ -522,6 +522,8 @@ and virtio_iso_path_matches_guest_os t path =
         (10, 0, not_client, ((=) "win2k19"))
       else if pathelem "2k22" || pathelem "win2022" then
         (10, 0, not_client, ((=) "win2k22"))
+      else if pathelem "2k25" || pathelem "win2025" then
+        (10, 0, not_client, ((=) "win2k25"))
       else
         raise Not_found in
 
diff --git a/lib/create_ovf.ml b/lib/create_ovf.ml
index f0b32e01..e2c19b08 100644
--- a/lib/create_ovf.ml
+++ b/lib/create_ovf.ml
@@ -242,10 +242,11 @@ and get_ostype = function
       | "win2k16" -> "windows_2016x64"
       | "win2k19" -> "windows_2019x64"
       | "win2k22" -> "windows_2022"
+      | "win2k25" -> "windows_2025"
       | _ ->
          warning (f_"unknown Windows 10 variant: %s (%s)")
            osinfo product;
-         "windows_2022"
+         "windows_2025"
      )
 
   | { i_type = typ; i_distro = distro;
@@ -448,6 +449,7 @@ and get_ovirt_osid = function
       | "win2k16" -> (* windows_2016x64 *) 29
       | "win2k19" -> (* windows_2019x64 *) 31
       | "win2k22" -> (* windows_2022 *) 37
+      (*| "win2k25" -> (* windows_2025 *) not yet known - 2024/10 *)
       | _ ->
          warning (f_"unknown Windows 10 variant: %s (%s)")
            osinfo product;
